effekt/shared/src/main/scala/effekt/machine/Transformer.scala

```diff
@@ -179,18 +179,34 @@ object Transformer {
         }
 
       case lifted.Match(scrutinee, clauses, default) =>
-        // TODO unordered matches
-        // TODO overlapping matches
-        // TODO incomplete matches
-        // TODO nested matches
-        transform(scrutinee).run { value => ???
-          //          Switch(value, clauses.map {
-          //            case (lifted.IgnorePattern(), _) => ??? // Clause(List(), NOOP) // TODO verify
-          //            case (lifted.AnyPattern(), _) => ???
-          //            case (lifted.TagPattern(constructor, patterns), lifted.BlockLit(params, body)) =>
-          //              Clause(params.map(transform), transform(body))
-          //            case (lifted.LiteralPattern(l), block) => ???
-          //          })
+        val constructors = clauses match {
+            case (symbols.Constructor(_, _, _, tpe), _) :: _ => tpe match {
+              case symbols.DataType(_, _, constructors) => constructors
+              case symbols.Record(_, _, constructor) => List(constructor)
+              case symbols.ExternType(_, _) => ??? // TODO: can this happen?
+            }
+         case _ => ??? // TODO: can there be an empty set of clauses? How do we get the type then?
+        };
+        transform(scrutinee).run { value =>
+          Switch(value, constructors.map {
+                   constructor => clauses.find { _._1 == constructor } match {
```

Comment:
This should not be necessary anymore, no?

Reply:
https://llvm.org/docs/LangRef.html#id39

```
switch i32 %val, label %otherwise [ i32 0, label %onzero
                                    i32 1, label %onone
                                    i32 2, label %ontwo ]
```

Reply:
The machine Switch just also needs a default label:

Currently:
```scala
case Switch(scrutinee: Variable, clauses: List[Clause])
```

Proposed:
```scala
case Switch(scrutinee: Variable, clauses: List[(Int, Clause)], default: Clause)
```
Note, that each clause also has an integer value.

Reply:
BTW, unsurprisingly, nested matching now also works in LLVM:

![image](https://user-images.githubusercontent.com/408265/201784314-14e608ae-4c20-464f-b9d9-778ff55a1525.png)